astropy/io/fits/hdu/compressed.py

```diff
@@ -105,9 +108,19 @@ class CompImageHeader(Header):
     # the schema system, but it's not quite ready for that yet.  Also it still
     # makes more sense to change CompImageHDU to subclass ImageHDU :/
 
-    def __init__(self, table_header, image_header=None):
+    def __new__(cls, table_header, image_header=None):
+        # 2019-09-14 (MHvK): No point wrapping anything if no image_header is
+        # given.  This happens if __getitem__ and copy are called - our super
+        # class will aim to initialize a new, possibly partially filled
+        # header, but we cannot usefully deal with that.
+        # TODO: the above suggests strongly we should *not* subclass from
```

Comment:
Puh, that's probably worth an issue. However the file includes also a few notes that suggest one should subclass `ImageHDU` for the compressed HDU. 

This is the kind of architectural change that probably shouldn't be put into `TODO` comments but discussed in the issue tracker.

Reply:
Agreed, there is #3895 about compressed HDU enhancements but it does mention this subclass issue. Could you open a new issue to track this ?

Reply:
See #9238